src/core/server/server.ts

```diff
@@ -550,4 +556,33 @@ export class Server {
       },
     });
   }
+
+  /**
+   * Reports the new and legacy KIBANA_STARTED_EVENT.
+   * @param analyticsStart The {@link AnalyticsServiceStart}.
+   * @private
+   */
+  private reportKibanaStartedEvents(analyticsStart: AnalyticsServiceStart) {
+    // Report the legacy KIBANA_STARTED_EVENT.
+    analyticsStart.reportEvent(KIBANA_STARTED_EVENT, { uptime_per_step: this.uptimePerStep });
+
+    const ups = this.uptimePerStep;
+
+    const toMs = (sec: number) => Math.round(sec * 100);
+    // Report the metric-shaped KIBANA_STARTED_EVENT.
+    reportMetricEvent(analyticsStart, {
+      eventName: KIBANA_STARTED_EVENT,
+      duration: toMs(ups.start!.end - ups.constructor!.start),
+      key1: 'time_to_constructor',
+      value1: toMs(ups.constructor!.start),
+      key2: 'constructor_time',
+      value2: toMs(ups.constructor!.end - ups.constructor!.start),
+      key3: 'preboot_time',
+      value3: toMs(ups.preboot!.end - ups.preboot!.start),
+      key4: 'setup_time',
+      value4: toMs(ups.setup!.end - ups.setup!.start),
+      key5: 'start_time',
+      value5: toMs(ups.start!.end - ups.start!.start),
+    });
```

Comment:
I just realized that we are actually missing a `key6/value6` at this point. A relevant metric to all this in Kibana is how many plugins Kibana is running with because it highly affects the startup time of the server.

Looking again at it, I still wonder the utility of these `keyN/valueN` metrics compared to a `flattened` type that can be referenced by a Runtime field if anyone wants to create any custom dashboard based on their own _free-form metrics_.

Comparing the _legacy_ event and the new one, I don't see a clear benefit other than having a standard `duration` field to have _One dashboard to rule them all_ and some PR/CI bots react to it (and even that field is optional 🤯 ).